src/editor/components/menu/editor-toolbar-menu.jsx

```diff
@@ -28,8 +32,10 @@ export class EditorToolbarMenuUnconnected extends React.Component {
 
 export function mapStateToProps(state) {
   const isServer = connectionModeIsServer(state);
+
   return {
-    isServer
+    isServer,
+    canUpload: isServer && Boolean(state.notebookInfo.user_can_save)
```

Comment:
Good catch here. One additional problem, though – if I fork the notebook, we need to make sure to clear the `files` object here. This is a more general problem in forking not specific to this. So let's file a follow-up issue after this is merged and we can address this problem.